src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs

```diff
@@ -2924,9 +2924,9 @@ public override IEnumerable<ISymbol> GetSymbolsWithName(Func<string, bool> predi
 
         #endregion
 
-        internal override AnalyzerDriver AnalyzerForLanguage(ImmutableArray<DiagnosticAnalyzer> analyzers, AnalyzerOptions options, Func<Exception, DiagnosticAnalyzer, bool> continueOnAnalyzerException, CancellationToken cancellationToken)
+        internal override AnalyzerDriver AnalyzerForLanguage(ImmutableArray<DiagnosticAnalyzer> analyzers, AnalyzerOptions options, AnalyzerManager analyzerManager, Func<Exception, DiagnosticAnalyzer, bool> continueOnAnalyzerException, CancellationToken cancellationToken)
```

Comment:
I think the term AnalyzerManager is probably a misnomer.
The newly added type is basically something that lives for the lifetime of analyzer host and ensures that individual hosts don't need to track if Initialize method/Compilation start actions/Supported Diagnostics etc. have been executed for an analyzer, and if so what are the cached results. 
This type is not intended to be the primary manager of everything related to analyzers, that is a pretty significant refactoring which doesn't buy us any new functionality and is likely going to change with John's work, so I am not going to attempt that here.
Do you have a better name suggestion for this type?